packages/wrangler/src/pages/deploy.tsx

```diff
@@ -73,38 +74,49 @@ export function Options(yargs: CommonYargsArgv) {
 		});
 }
 
-export const Handler = async ({
-	_,
-	directory,
-	projectName,
-	branch,
-	commitHash,
-	commitMessage,
-	commitDirty,
-	skipCaching,
-	bundle,
-	noBundle,
-	config: wranglerConfig,
-}: PublishArgs) => {
+export const Handler = async (args: PublishArgs) => {
+	let { branch, commitHash, commitMessage, commitDirty } = args;
+
 	// Check for deprecated `wrangler pages publish` command
-	if (_[1] === "publish") {
+	if (args._[1] === "publish") {
 		logger.warn(
 			"`wrangler pages publish` is deprecated and will be removed in the next major version.\nPlease use `wrangler pages deploy` instead, which accepts exactly the same arguments."
 		);
 	}
 
-	if (wranglerConfig) {
-		throw new FatalError("Pages does not support wrangler.toml", 1);
+	if (args.config) {
+		throw new FatalError(
+			"Pages does not support custom paths for the `wrangler.toml` configuration file",
+			1
+		);
 	}
 
+	// TODO Carmen
+	// this reads the config file with `env` set to undefined, which will
```

Comment:
in the context of my question below about diff project `name`s per environment, I'm wondering how reading the config properly would work 🤔 . 

In order to read the config for a specific env, we need to pass the environment name. For Pages, detecting the environment name is done here:
```
const project = await fetchResult<Project>(
		`/accounts/${accountId}/pages/projects/${projectName}`
	);
let isProduction = true;
if (branch) {
	isProduction = project.production_branch === branch;
}

const deploymentConfig = project.deployment_configs[isProduction ? "production" : "preview"];
```
That means we need the project name to make that API call. But the project name is potentially coming from the config file, and if we support diff project names per env, we would need the environment name to get the project name 😅 . Soooo...that would be like a circular dependency situation? Unless I'm completely misguided in my reasoning

Reply:
We don't support different project names per environment (unlike Workers), so this should be fine, I think?

Reply:
yes, then no circular dependencies for us 😋